Extended Paperless-ngx docs (#40012)
home-assistant/home-assistant.github.io: source/_integrations/paperless_ngx.markdown

| **Status sanity**        | Indicates the sanity of the Paperless-ngx documents.                           |
| **Software**             | Indicates whether a new Paperless-ngx update ist available.                    |

## Example automations

{% details "Send a push notification if a new document is available" %}
{% raw %}

```yaml
alias: New document push notification
description: Sends a push notification if a new document is available
triggers:
  - entity_id: sensor.paperless_documents_inbox
    to: null
    trigger: state
conditions:
  - condition: template
    value_template: |
      {% if trigger.from_state is not none and trigger.to_state is not none %}
        {{ trigger.to_state.state > trigger.from_state.state }}
      {% else %}
        false
      {% endif %}
actions:
  - action: notify.mobile_app_iphone
    metadata: {}
    data:
      message: A new document is available.
```

{% endraw %}
{% enddetails %}

> ## Data Updates

Reviewer: ```suggestion
## Data updates
```

This integration retrieves data using a pull-based mechanism.

- **Statistic sensors** are pulled every **120 seconds**  
- **Diagnostic sensors** are pulled every **300 seconds**  
- **Update checks** to detect new Paperless-ngx versions are performed **every 24 hours**

## Known limitations

There are a few known limitations for using the integration:

- This integration is only fully supported with **Paperless-ngx version 2.15 or later**. Earlier versions are not supported.
- To enable monitoring of diagnostic sensors, you must have **administrator permissions**. Without administrator rights, specific API endpoints cannot be accessed, and the sensor states will not be available.

## Removing the integration

This integration follows standard integration removal. No extra steps are required.

{% include integrations/remove_device_service.md %}

## Troubleshooting

> {% details "Invalid hostname or IP address" %}

Reviewer: Is this a message I see? or the cause of an issue?

> {% details "If you get the message 'Invalid hostname or IP address', try the following steps:"" %}
> 

Reviewer: ```suggestion
{% details "Message: 'Invalid hostname or IP address'" %}

If you get the message **Invalid hostname or IP address**, try the following steps:
```
Sorry, my comment was a bit unclear. I meant in the intro, not the title. A short title, like you had it before, is good.

Author: Ah okay, got it and thanks for the changes :)

Make sure you enter a complete URL, such as `https://paperless.example.com` or `https://192.168.178.11:8011`.

SSL is enabled by default. If you're using an unencrypted connection, you must explicitly use `http://` instead of `https://` in the URL.

If you're using a self-signed certificate, disable the **Verify SSL certificate** option.

{% enddetails %}

> {% details "The token does not have permission to access the API" %}

Reviewer: same here. Is this an error messsage that comes up or the cause of an issue?

Author: It's an error message. Should I show it with a prefix like 'Error message: [...]'?

Reviewer: I just think it's important the symptom is clearly mentioned.
maybe sth like
 If you get the message "The token does not have permission to access the API", try the following steps:

> {% details "If you get the message 'The token does not have permission to access the API', try the following steps:" %}
> 

Reviewer: ```suggestion
{% details "Message: 'The token does not have permission to access the API'" %}

If you get the message **The token does not have permission to access the API**, try the following steps:
```

In this case, verify whether the token is still valid and correctly assigned to the user.

You can also test the token using the Swagger interface available at
`https://paperless.example.com/api/schema/view/`

Click on **"Authorize"** in the Swagger UI to enter your token at **tokenAuth (apiKey)**, then try accessing the relevant endpoints like `/api/statistics/` to ensure they respond as expected.

> If everything works correctly in Swagger but the integration still fails, check whether a reverse proxy (e.g., NGINX) is returning an **HTTP 403 error**. If so, the integration may also report this as a permission issue.

Reviewer: ```suggestion
1. Verify whether the token is still valid and correctly assigned to the user.
2. You can also test the token using the Swagger interface available at
`https://paperless.example.com/api/schema/view/`
   - In the Swagger UI, select **Authorize** to enter your token at **tokenAuth (apiKey)**.
   - Then, try accessing the relevant endpoints like `/api/statistics/` to ensure they respond as expected.
3. If everything works correctly in Swagger but the integration still fails, check whether a reverse proxy (for example, NGINX) is returning an **HTTP 403 error**. 
   - If so, the integration may also report this as a permission issue.
```
I think this could be a step-by-step procedure

> {% enddetails %}

Reviewer: ```suggestion
{% enddetails %}

## Removing the integration

This integration follows standard integration removal, no extra steps are required.

{% include integrations/remove_device_service.md %}
```
Would this section also apply for your integration? Or are there extra steps required, like removing some data or an account or something?

Author: Yes, this applies for my integration. The default steps are enough.
